solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/filter/SetAccessStatusFilter.java

```diff
@@ -105,13 +108,45 @@ public void setObjectAclFactory(ObjectAclFactory oaf) {
             status.add(FacetConstants.PUBLIC_ACCESS);
         }
 
-        if (objPrincRoles.size() > 0) {
+        if (addPatronSettings(dip.getContentObject())) {
             status.add(FacetConstants.PATRON_SETTINGS);
         }
 
         return status;
     }
 
+    private boolean addPatronSettings(ContentObject contentObj) {
+        List<String> objTypes = contentObj.getTypes();
+        ResourceType objType = ResourceType.getResourceTypeForUris(objTypes);
+        if (objType == null || objType.equals(Cdr.AdminUnit.getURI())) {
+            return false;
+        }
+
+        return hasPatronSettings(contentObj.getPid(), objType);
+    }
+
+    private boolean hasPatronSettings(PID pid, ResourceType rescType) {
+        List<RoleAssignment> objPatronRoles = objAclFactory.getPatronRoleAssignments(pid);
+        int numRoles = objPatronRoles.size();
+        boolean isCollection = rescType.equals(Cdr.Collection.getLocalName());
+
+        if ((!isCollection && numRoles == 0)) {
+            return false;
+        } else if (isCollection && numRoles < 2) {
+            return true;
+        }
+
+        String canViewOrig = canViewOriginals.getURI().toString();
+        UserRole roleOne = objPatronRoles.get(0).getRole();
```

Comment:
I'd suggest iterating through the list of roles and returning false if any of them aren't canViewOriginal. There will probably be more than 2 groups in the future, and should have the same affect.